### ISSUE TITLE
Restarting the VCH, imagec tries to create the image store directory when it already exists

### ISSUE THREAD

**Issue by fdawg4l** [vmware/vic#491]
```
time="2016-04-08T23:33:45Z" level=info msg="[BEGIN] [main.CreateImageStore] Photon" 
time="2016-04-08T23:33:45Z" level=debug msg="Creating a store" 
time="2016-04-08T23:33:46Z" level=debug msg="Creating a store failed: [POST /storage][500] CreateImageStore default  &{Code:0xc82035bf60 Message:ServerFaultCode: Cannot complete the operation because the file or folder [datastore1] VIC/Photon already exists}" 
time="2016-04-08T23:33:46Z" level=info msg="[ END ] [main.CreateImageStore] [15.943048ms] Photon" 
time="2016-04-08T23:33:46Z" level=fatal msg="Failed to create image store: [POST /storage][500] CreateImageStore default  &{Code:0xc82035bf60 Message:ServerFaultCode: Cannot complete the operation because the file or folder [datastore1] VIC/Photon already exists}" 
```

I did a `docker pull golang:1.6` on a fresh ESX+VCH, let it complete, stopped the VCH and restarted it, then issued it again.  The port-layer correctly returned the EEXIST error.  Imagec should swallow this and move on.

**Comment by fdawg4l** [comment]
`ListImages` still isn't implemented on the portlayer REST side or the portlayer storage side.  The stub exists in the vsphere datastore side, but nothing is calling it.  I'll create an issue for this if one doesn't already exist.  In any case, restart won't work without `ListImages` actually going down to the datastore and populating the cache and the REST side returning it.

**Comment by fdawg4l** [comment]
REST API issue is captured by #114 
SPL change is captured by #298

**Comment by caglar10ur** [comment]
Something else must be changed on the post-layer-server side as imagec already does what this issue describes as long as port-layer-returns expected answer.

```
 60     if _, ok := err.(*storage.CreateImageStoreConflict); ok {
 61         log.Debugf("Store already exists")
 62         return nil
 63     }
 64     if err != nil {
 65         log.Debugf("Creating a store failed: %s", err)
 66
 67         return err
 68     }
```

@jzt did something change recently on the server side?

@fdawg4l is this happening on master?

**Comment by fdawg4l** [comment]
I missed a crucial step in the description.  I restarted the VCH after doing the pull.  Apologies.

@caglar10ur no, not on master.  This is with my change (#495).

**Comment by fdawg4l** [comment]
Looks like the CreatedImageStoreConflict isn't being returned as part of my change. Thanks for looking into it @caglar10ur!  I'll take this issue and fix the returned error so we can support this.

**Comment by fdawg4l** [comment]
Fixed in a3ec0374c823576335cb7a5cff1acdb40dfb3aa9.
